test/modules/TestResultCoordinator/Reports/DirectMethod/DirectMethodReportMetadata.cs

```diff
@@ -2,26 +2,45 @@
 namespace TestResultCoordinator.Reports.DirectMethod
 {
     using System;
+    using System.Collections.Generic;
     using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Newtonsoft.Json;
     using TestResultCoordinator.Reports;
 
     class DirectMethodReportMetadata : ITestReportMetadata
     {
-        public DirectMethodReportMetadata(string expectedSource, string actualSource, TestReportType testReportType, TimeSpan tolerancePeriod)
+        public DirectMethodReportMetadata(string senderSource, TestReportType testReportType, TimeSpan tolerancePeriod, string receiverSource = "")
         {
-            this.ExpectedSource = expectedSource;
-            this.ActualSource = actualSource;
+            this.SenderSource = senderSource;
             this.TestReportType = testReportType;
             this.TolerancePeriod = tolerancePeriod;
+            if (receiverSource is null || string.Empty.Equals(receiverSource))
+            {
+                this.ReceiverSource = Option.None<string>();
+            }
+            else
+            {
+                this.ReceiverSource = Option.Some(receiverSource);
+            }
         }
 
         public TimeSpan TolerancePeriod { get; }
 
-        public string ExpectedSource { get; }
+        public string SenderSource { get; }
 
-        public string ActualSource { get; }
+        public Option<string> ReceiverSource { get; }
 
-        public string[] ResultSources => new string[] { this.ExpectedSource, this.ActualSource };
+        public string[] ResultSources
+        {
+            get
+            {
+                List<string> resultSources = new List<string>();
+                resultSources.Add(this.SenderSource);
```

Comment:
For everytime `ResultSources` is accessed, add a `SenderSource` to the `resultSources`? I'm confused.

Reply:
Changed around to make it more readable